React/Views/RCTScrollView.m

```diff
@@ -317,8 +317,8 @@ - (void)setFrame:(CGRect)frame
   CGSize boundsSize = self.bounds.size;
 
   self.contentOffset = CGPointMake(
-    MAX(0, MIN(originalOffset.x, fullContentSize.width - boundsSize.width)),
-    MAX(0, MIN(originalOffset.y, fullContentSize.height - boundsSize.height)));
+    MAX(-contentInset.left, MIN(originalOffset.x, fullContentSize.width - boundsSize.width)),
```

Comment:
@shergin I looked at the issue with the `RCTScrollView#setFrame` code and it seems worse than I thought. What happens is that when `setFrame` is called initially the content view isn't measured yet so `self.contentSize` has size {0,0}. This makes it so setting the contentOffset prop has no effect on initial render since initial offset will always be {0,0} (or -contentInset.{top|left} with the fix from this PR). I'll let you have a look since I'm not sure what was the bug the changes in `setFrame` were supposed to fix.